rest-api-spec/src/main/resources/rest-api-spec/test/indices.create/20_mix_typeless_typeful.yml

```diff
@@ -0,0 +1,77 @@
+---
+"Create a typeless index while there is a typed template":
```

Comment:
Since the change is a bit subtle, I wonder if it'd be worth adding some unit tests in `IndexCreationTaskTests`? In particular, reading the code I had to think a bit about what would happen if no mappings were provided in the create index requests, or if the template or create request had more than one type.